src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +229,160 @@ def storage(request):
+
+    # Calculate usage/size, if requested
+    if calculate_usage:
+        for directory in dirs:
+            if dirs[directory].get('contained_by') is None:
+               # Get size/usage of space
+                space_path = dirs[directory]['path']
+                dirs[directory]['size'] = _usage_check_directory_volume_size(space_path)
+                dirs[directory]['used'] = _usage_check_directory_used(space_path)
+            else:
+                # Get size from containing space and calculate usage of specific path
+                space = dirs[directory]['contained_by']
+                dirs[directory]['size'] = dirs[space]['size']
+                dirs[directory]['used'] = _usage_check_directory_used(dirs[directory]['path'])
+
+    return dirs
+
+def _usage_check_directory_volume_size(path):
+    # Get volume size (in 512 byte blocks)
+    try:
+        output = subprocess.check_output(["df", path])
+        device, size, used, available, percent, mountpoint = output.split("\n")[1].split()
+        return int(size) * 512
+    except subprocess.CalledProcessError:
+        return 0
+
+def _usage_check_directory_used(path):
+    # Get total usage in bytes
+    try:
+        output = subprocess.check_output(["du", "--bytes", "--summarize", path])
+        return output.split("\t")[0]
+    except subprocess.CalledProcessError:
+        return 0
+
+def _usage_shared_dir_path():
+    # Get shared directory path
+    clientConfigFilePath = os.path.join(os.sep, 'etc', 'archivematica', 'MCPClient', 'clientConfig.conf')
+    config = ConfigParser.SafeConfigParser()
+    config.read(clientConfigFilePath)
+
+    return config.get('MCPClient', 'sharedDirectoryMounted')
+
+def _usage_location_path(purpose):
+    # Get currently processing location
+    locations = storage_service.get_location(purpose=purpose)
+    location = locations[0]
+    return location['path']
+
+def clear_context(request, dir_id):
+    usage_dirs = _usage_dirs(False)
+    prompt = 'Clear ' + usage_dirs[dir_id]['description'] + '?'
+    cancel_url = reverse("components.administration.views.usage")
+    return RequestContext(request, {'action': 'Delete', 'prompt': prompt, 'cancel_url': cancel_url})
+
+@user_passes_test(lambda u: u.is_superuser, login_url='/forbidden/')
+@decorators.confirm_required('simple_confirm.html', clear_context)
+def usage_clear(request, dir_id):
+    if request.method == 'POST':
+        usage_dirs = _usage_dirs(False)
+        dir_info = usage_dirs[dir_id]
+
+        # Prevent shared directory from being cleared
+        if dir_id == 'shared' or not dir_info:
+            raise Http404
+
+        # Determine if specific subdirectories need to be cleared, rather than
+        # whole directory
+        if dir_info.get('subdirectories') is not None:
+            dirs_to_empty = []
+            for subdirectory in dir_info['subdirectories']:
+                dirs_to_empty.append(os.path.join(dir_info['path'], subdirectory))
```

Comment:
Have you discovered the magic of [list comprehensions](https://docs.python.org/2/tutorial/datastructures.html#list-comprehensions)?  They're fantastic.  This could be `dirs_to_empty = [os.path.join(dir_info['path'], subdir) for subdir in dir_info['subdirectories']]`